src/shared/util.js

```diff
@@ -401,99 +401,54 @@ function shadow(obj, prop, value) {
   return value;
 }
 
-const PasswordException = (function PasswordExceptionClosure() {
-  function PasswordException(msg, code) {
-    this.name = 'PasswordException';
-    this.message = msg;
-    this.code = code;
+const BaseException = (function BaseExceptionClosure() {
```

Comment:
Once extending `Error` isn't a problem any more, when SystemJS has been updated/replaced, it should possible to simplify this even further to e.g.
```javascript
class BaseException extends Error {
  constructor(message) {
    if (this.constructor === BaseException) {
      unreachable('Cannot initialize BaseException.');
    }
    super(message);
    this.name = this.constructor.name;
  }
}
```